sollumz_operators.py

```diff
@@ -564,6 +546,43 @@ def parse_location_string(location_string):
             self.report({'ERROR'}, "Invalid location string.")
 
         return {'FINISHED'}
+    
+class SOLLUMZ_OT_paste_rotation(bpy.types.Operator):
+    """Paste the rotation (as quaternion) of an object from the clipboard and apply it"""
+    bl_idname = "wm.sollumz_paste_rotation"
+    bl_label = "Paste Rotation"
+    rotation: bpy.props.StringProperty()
+
+    def execute(self, context):
+
+        rotation_string = bpy.context.window_manager.clipboard
+
+        rotation_quaternion = parse_rotation_string(rotation_string)
+        if rotation_quaternion is not None:
+            selected_object = bpy.context.object
+
+            rotation_matrix = rotation_quaternion.to_matrix().to_4x4()
+            
+            selected_object.matrix_world @= rotation_matrix
```

Comment:
Here you are rotating the object by the pasted rotation, not setting its rotation.

https://github.com/Skylumz/Sollumz/assets/12873137/a7460c83-6f90-4663-9fb8-370380a57a4e

One approach I can think of to fix this is decomposing `matrix_world` with `Matrix.decompose()` and reconstruct the matrix with `Matrix.LocRotScale()`, but replacing the rotation quaternion with the one pasted in.